aea/decision_maker/base.py

```diff
@@ -165,22 +169,20 @@ def update(self, tx_message: TransactionMessage) -> None:
         :param tx_message:
         :return: None
         """
-        if tx_message.is_sender_buyer:
-            diff = tx_message.amount + tx_message.sender_tx_fee
-            self._amount_by_currency[tx_message.currency_id] -= diff
-        else:
-            diff = tx_message.amount - tx_message.sender_tx_fee
-            self._amount_by_currency[tx_message.currency_id] += diff
+        assert self.check_transaction_is_consistent(tx_message), "Inconsistent transaction."
+        for currency_id, amount in tx_message.tx_amount_by_currency_id.items():
+            # we apply the worst case where all the tx_sender_fee is used up in the transaction
```

Comment:
Can it be different? That is, for some transactions the agent might not be supposed to pay (because it's not the sender)? Or because the fee was too high than actually needed?

Reply:
There is a wider issue around transaction fees. The first is that the split of the `max_tx_fee` needs to be part of the negotiation. The actual `tx_fee` is only known at the time of transacting. The update() method here is actually a bit of a placeholder. Currently, it is used for forward looking calculations by the AEA, i.e. what happens if I transact like this to my overall utility. Actual state is updated via the `apply_state_update` method, which assumes that tx fees are roled into the `amount_by_currency_id` diff. This only gets used by the TAC skills atm. There is stilll a lot of work to be done to align the state the agent holds with the ledger state and come up with the right synchronisation.